### PR TITLE
Utility proxy

### PR DESCRIPTION
**Pull request by sandrampeter**
## Background

[proxy utility for AWS](https://app.asana.com/0/1200681690870921/1201744486584882/f)

### CODE REVIEW

File: fixtures/test_proxy/main.tf [hunk 1]
```diff
@@ -59,25 +65,26 @@ resource "aws_instance" "proxy" {
...
     volume_type = "gp2"
     volume_size = "100"
   }
 }
+
+module "test_proxy_init" {
+  source = "github.com/hashicorp/terraform-random-tfe-utility//fixtures/test_proxy_init?ref=main"
+
+  mitmproxy_ca_certificate_secret = var.mitmproxy_ca_certificate_secret != null ? var.mitmproxy_ca_certificate_secret : ""
+  mitmproxy_ca_private_key_secret = var.mitmproxy_ca_private_key_secret != null ? var.mitmproxy_ca_private_key_secret : ""
```
Comment (anniehedgpeth) [inline fixtures/test_proxy/main.tf:88]: Instead of `""` can we use `null`?

File: fixtures/test_proxy/variables.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  The identifier of a secret comprising a Base64 encoded PEM private key file for the mitmproxy Certificate Authority.
+  EOD
+  type        = string
+}
+
+# Network
+# -------
+
+variable "subnet_id" {
+  default     = ""
```
Comment (anniehedgpeth) [inline fixtures/test_proxy/variables.tf:24]: We want to use `null` as opposed to empty strings for defaults.
```suggestion
  default     = null
```

File: fixtures/test_proxy/variables.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  description = "The identity of the VPC in which resources will be deployed."
+  type        = string
+}
+
+variable "vpc_id" {
+  default     = ""
```
Comment (anniehedgpeth) [inline fixtures/test_proxy/variables.tf:30]:
```suggestion
  default     = null
```
Same here.

File: fixtures/test_proxy/variables.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  type        = string
+}
+
+# PROXY SETTINGS
+# --------------
+
+variable "name" {
+  type        = string
+  description = "Name of the proxy server."
+  default     = ""
```
Comment (anniehedgpeth) [inline fixtures/test_proxy/variables.tf:41]:
```suggestion
  default     = null
```

**Review by anniehedgpeth — CHANGES_REQUESTED** [review]
Looks good other than using empty strings as default variable values. They can sometimes obfuscate issues, so we'd rather use `null`. (For context, we created this standard [back here](https://app.asana.com/0/1181500399442529/1200487922889435/f).)

**Comment by anniehedgpeth** [comment]
Thanks, @sandrampeter ! Would you mind testing the proxies once more just to make sure there's no unexpected behavior from that change?

**Comment by sandrampeter** [comment]
> Thanks, @sandrampeter ! Would you mind testing the proxies once more just to make sure there's no unexpected behavior from that change?

@anniehedgpeth its done

**Review by anniehedgpeth — APPROVED** [review]

**Pull request merged**